modules/tensor_mechanics/include/materials/LAROMANCEStressUpdateBase.h

```diff
@@ -526,6 +530,17 @@ class LAROMANCEStressUpdateBaseTempl : public RadialReturnCreepStressUpdateBaseT
 
   /// Unit conversion factors required to convert from the specified unit to MPa
   Real _stress_ucf;
+
+  ///@{Material properties accumulated at substeps
+  GenericMaterialProperty<Real, is_ad> & _wall_dislocations_step;
+  GenericMaterialProperty<Real, is_ad> & _cell_dislocations_step;
+  ///@}
+
+  /// Total plastic strain increment in step (summing substep contributions)
+  RankTwoTensor _plastic_strain_increment;
+
+  /// Material capturing number of substeps. If substepping isn't used will default to one.
+  MaterialProperty<Real> & _number_of_substeps;
```

Comment:
why is it `Real` type? not `unsigned int`

Reply:
Should be `unsigned int` but the materialOutputAction somehow can't handle it. It gives me an error: 

``` *** ERROR ***
test:rom_stress_update.ADrom/lower_limit_extrapolated: The types for total 1 material properties:
test:rom_stress_update.ADrom/lower_limit_extrapolated:   number_of_substeps
test:rom_stress_update.ADrom/lower_limit_extrapolated: are not supported for automatic output by MaterialOutputAction. *** ERROR ***
```
There are very few instances of `MaterialProperty<unsigned int>` in MOOSE.

Reply:
That is wired. My suggestion is either to make `unsigned int` work with `materialOutputAction` or disable material output for the number of substeps using `materialOutputAction`. You can manually output substeps using an auxkernel.